frontends/web/src/utils/request.ts

```diff
@@ -35,18 +36,31 @@ function extConfig(key, defaultValue) {
 export const apiPort = extConfig('{{ API_PORT }}', '8082');
 export const apiToken = extConfig('{{ API_TOKEN }}', '');
 
-export function isTLS() {
+export function isTLS(): boolean {
   return document.URL.startsWith('https://');
 }
 
-export function apiURL(endpoint) {
+export function apiURL(endpoint: string): string {
   return (isTLS() ? 'https://' : 'http://') + 'localhost:' + apiPort + '/api/' + endpoint;
 }
 
-function handleError(endpoint) {
-  return function(json) {
+// https://fettblog.eu/typescript-hasownproperty/
+function hasOwnProperty<X extends {}, Y extends PropertyKey>(
+  obj: X,
+  prop: Y,
+): obj is X & Record<Y, unknown> {
+  return obj.hasOwnProperty(prop);
+}
+
+function handleError(endpoint: string) {
+  return function(json: unknown) {
     return new Promise((resolve, reject) => {
-      if (json && json.error) {
+      if (
+        typeof json === 'object'
+        && json !== null
```

Comment:
is this needed? if it is an object (previous line), it is not null, right?

Reply:
yes seems so, lets check together…

![Screen Shot 2022-10-05 at 14 21 59](https://user-images.githubusercontent.com/546900/194059261-cd54c729-0b12-45e0-a671-6314e95bad37.png)